pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java

```diff
@@ -240,7 +240,9 @@ protected void updateTopicPolicy(TopicPolicies data) {
         topicPolicies.getDispatchRate().updateTopicValue(DispatchRateImpl.normalize(data.getDispatchRate()));
         topicPolicies.getSchemaValidationEnforced().updateTopicValue(data.getSchemaValidationEnforced());
         topicPolicies.getEntryFilters().updateTopicValue(data.getEntryFilters());
-        this.subscriptionPolicies = data.getSubscriptionPolicies();
+        if (data.getSubscriptionPolicies() != null) {
```

Comment:
Should we check line-174 instead of this line?

https://github.com/apache/pulsar/blob/1e44ba179bfc189d22dff046a1887e12842b0851/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java#L173-L178

Reply:
`subscriptionPolicies` shouldn't be null, only `updateTopicPolicy ` may change it, so I think it's better to modify line-243.

Reply:
I am wondering why `subscriptionPolicies` can be null.  Is it passed by `TopicPoliceis`? Because I saw many places invoke `TopicPoliceis#getSubscriptionPolicies.get(xxx)` without an NPE check. 
Should we consider avoiding this value as null? 

Please correct me if I got something wrong.

/cc @Technoboy- @poorbarcode